modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/WeightedSampling.scala

```diff
@@ -0,0 +1,43 @@
+package org.tessellation.sdk.infrastructure.snapshot
+
+import cats.MonadThrow
+import cats.syntax.applicative._
+import cats.syntax.applicativeError._
+import cats.syntax.either._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.Interval
+
+object WeightedSampling {
+
+  type ProbabilityRefinement = Interval.ClosedOpen[0.0, 1.0]
+  type Probability = Double Refined ProbabilityRefinement
+
+  type WeightRefinement = Interval.OpenClosed[0.0, 1.0]
+  type Weight = Double Refined WeightRefinement
+
+  def sample[F[_]: MonadThrow, A](chosenP: Probability)(distribution: List[(A, Weight)]): F[A] = {
```

Comment:
can you rename the argument `chosenP` to `probability`?
so the definition would look like:
```
def sample[F[_]: MonadThrow, A](probability: Probability)(distribution: List[(A, Weight)]): F[A]
```

Reply:
Changed `chosenP` to `probability`.